app/lib/experiments-client.coffee

```diff
@@ -0,0 +1,206 @@
+config = require './experiments-config'
+{getSessionID} = require '../lib/session'
+DEBUG = false
+
+class ExperimentsClient # Client for the ExperimentServer
+
+  EXPERIMENT_SERVER_PRODUCTION_URL: "https://experiments.zooniverse.org"
+  EXPERIMENT_SERVER_STAGING_URL: "https://experiments.staging.zooniverse.org"
+  #EXPERIMENT_SERVER_DEVELOPMENT_URL: "http://localhost:4567"
+  EXPERIMENT_SERVER_URL_TO_USE: null
+  
+  ensureEnvironmentSet: ->
+    if not @EXPERIMENT_SERVER_URL_TO_USE?
+      if window.location.hostname=="www.zooniverse.org"
+        @EXPERIMENT_SERVER_URL_TO_USE = @EXPERIMENT_SERVER_PRODUCTION_URL
+      #else if window.location.hostname=="localhost"
+      #  @EXPERIMENT_SERVER_URL_TO_USE = @EXPERIMENT_SERVER_DEVELOPMENT_URL
+      else
+        @EXPERIMENT_SERVER_URL_TO_USE = @EXPERIMENT_SERVER_STAGING_URL
+    @EXPERIMENT_SERVER_URL_TO_USE
+  
+  getEnabledExperiments: ->
+    @ensureEnvironmentSet()
+    switch @EXPERIMENT_SERVER_URL_TO_USE
+      when @EXPERIMENT_SERVER_PRODUCTION_URL
+        config.ENABLED_EXPERIMENTS_PRODUCTION
+      else
+        config.ENABLED_EXPERIMENTS_STAGING
+
+  getInterventionDetails: ->
+    @ensureEnvironmentSet()
+    switch @EXPERIMENT_SERVER_URL_TO_USE
+      when @EXPERIMENT_SERVER_PRODUCTION_URL
+        config.INTERVENTION_DETAILS_PRODUCTION
+      else
+        config.INTERVENTION_DETAILS_STAGING
+
+  getProjectSlugs: ->
+    @ensureEnvironmentSet()
+    switch @EXPERIMENT_SERVER_URL_TO_USE
+      when @EXPERIMENT_SERVER_PRODUCTION_URL
+        config.PROJECT_SLUGS_PRODUCTION
+      else
+        config.PROJECT_SLUGS_STAGING
+
+  lookupProjectSlug: (enabled_experiments, experiment_name) ->
+    this_project_slug for this_project_slug, experiments_for_this_slug of enabled_experiments when experiment_name in experiments_for_this_slug
```

Comment:
Good point. I've been switching between Python, Ruby and CoffeeScript so much for the different components of this and sometimes forget where I am!